node/src/components/chainspec_loader.rs

```diff
@@ -293,8 +285,30 @@ impl ChainspecLoader {
         self.next_upgrade.clone()
     }
 
+    pub(crate) fn highest_block_header(&self) -> Option<&BlockHeader> {
+        self.highest_block_header.as_ref()
+    }
+
     pub(crate) fn highest_block_hash(&self) -> Option<BlockHash> {
-        self.highest_block_hash
+        self.highest_block_header().map(|hdr| hdr.hash())
+    }
+
+    pub(crate) fn starting_era(&self) -> EraId {
```

Comment:
Can you add a comment on the method itself about what it returns and in which case?